lib/autocomplete-manager.coffee

```diff
@@ -1,139 +1,140 @@
-      "autocomplete-plus:confirm": @confirmSelection,
-      "autocomplete-plus:select-next": @selectNext,
-      "autocomplete-plus:select-previous": @selectPrevious,
-      "autocomplete-plus:cancel": @cancel
-
-  addKeyboardInteraction: ->
-    @removeKeyboardInteraction()
-    keys =
-      "escape": "autocomplete-plus:cancel"
-
-    completionKey = atom.config.get("autocomplete-plus.confirmCompletion") || ''
-    navigationKey = atom.config.get("autocomplete-plus.navigateCompletions") || ''
-
-
-    keys['tab'] = "autocomplete-plus:confirm" if completionKey.indexOf('tab') > -1
-    keys['enter'] = "autocomplete-plus:confirm" if completionKey.indexOf('enter') > -1
-
-    if @items?.length > 1 and navigationKey == "up,down"
-      keys['up'] =  "autocomplete-plus:select-previous"
-      keys['down'] = "autocomplete-plus:select-next"
-    else
-      keys["ctrl-n"] = "autocomplete-plus:select-next"
-      keys["ctrl-p"] = "autocomplete-plus:select-previous"
-
-    @keymaps = atom.keymaps.add(
-      'AutocompleteManager',
-      'atom-text-editor:not(.mini) .autocomplete-plus': keys
-    )
-
-    @compositeDisposable.add @keymaps
-
-  removeKeyboardInteraction: ->
-    @keymaps?.dispose()
-    @compositeDisposable.remove(@keymaps)
+    @handleCommands()
+    @fuzzyProvider = new FuzzyProvider()
 
   updateCurrentEditor: (currentPaneItem) =>
-    @cancel() unless currentPaneItem == @editor
-
-  confirmSelection: =>
-    @emitter.emit 'do-confirm-selection'
-
-  onDoConfirmSelection: (cb) ->
-    @emitter.on 'do-confirm-selection', cb
-
-  selectNext: =>
-    @emitter.emit 'do-select-next'
-
-  onDoSelectNext: (cb) ->
-    @emitter.on 'do-select-next', cb
+    return unless currentPaneItem?
+    return if currentPaneItem is @editor
+
+    # Stop listening to buffer events
+    @bufferSavedSubscription?.dispose()
+    @bufferChangedSubscription?.dispose()
+
+    # Stop listening to editor events
+    @editorCursorMovedSubscription?.dispose()
+    @didChangeTabsSubscription?.dispose()
+
+    # Disqualify invalid pane items
+    # TODO
+    if false
+      @editor = null
+      @editorView = null
+      @buffer = null
+
+    # Track the new editor, editorView, and buffer
+    @editor = currentPaneItem
+    @editorView = atom.views.getView(@editor)
+    @buffer = @editor.getBuffer()
 
-  selectPrevious: =>
-    @emitter.emit 'do-select-previous'
+    # Subscribe to buffer events:
+    @bufferSavedSubscription = @buffer.onDidSave(@bufferSaved)
+    @bufferChangedSubscription = @buffer.onDidChange(@bufferChanged)
 
-  onDoSelectPrevious: (cb) ->
-    @emitter.on 'do-select-previous', cb
+    # Subscribe to editor events:
+    # Close the overlay when the cursor moved without changing any text
+    @editorCursorMovedSubscription = @editor.onDidChangeCursorPosition(@cursorMoved)
+    # TODO: Is this the event for switching tabs?
+    @didChangeTabsSubscription = @editor.onDidChangeTitle(@hideSuggestionList)
```

Comment:
i really think you can probably take this event out, as long as we are detaching from the editors when they are no longer active this should not be necessary.